### ISSUE TITLE
HighDPI support needs tuning

### ISSUE THREAD

**Issue by pawelsalawa** [pawelsalawa/sqlitestudio#3987]
More details in #3046

**Comment by pawelsalawa** [comment]
@regs01 How about this approach? Does it make any difference for you?
![image](https://user-images.githubusercontent.com/481846/108685971-169b0600-74f5-11eb-8b18-6e115c4b94c6.png)

**Comment by pawelsalawa** [comment]
Info for myself - to test under VBox with Windows guest - need to run from command line:
`"C:\Program Files\Oracle\VirtualBox\VBoxManage.exe" controlvm "MSEdge - Win10" setvideomodehint 3400 2100 32`
and set "scaled" view in VB guest window.

**Comment by pawelsalawa** [comment]
Reference docs: https://doc.qt.io/qt-5/highdpi.html

**Comment by regs01** [comment]
**Application** is when system is reporting to application real screen DPI, so application can scale **itself** to desired DPI. This is the way we talking about. I.e. If you have 150% scaling, system will report to application 144 DPI. Qt has dpiawareness = 1 (one for all monitors) or 2 (per monitor) for that.

**System and System (Enhanced)** is when system is reporting 96 DPI to application, but applying rasterized bitmap scaling to the window. Meaning the app don't have to deal with scaling itself. System will increase/decrease the window with all its content. But does it as a bitmap, so the window will appear very blurry. This is a compatibility option for very older applications that have no scaling capabilities. That is dpiawareness = 0 in Qt.

Older Windows 10 versions also had a way to disable scaling. This way the system will report 96 DPI to application and won't do anything to it. And so as application won't do anything, as it's thinking there is no need to scale anything. It was removed in newer Windows 10 versions as pointless.

**Comment by regs01** [comment]
This is qBittorrent approach.
```
#if !defined(DISABLE_GUI) && (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
    // Attribute Qt::AA_EnableHighDpiScaling must be set before QCoreApplication is created
    if (qgetenv("QT_ENABLE_HIGHDPI_SCALING").isEmpty() && qgetenv("QT_AUTO_SCREEN_SCALE_FACTOR").isEmpty())
        Application::setAttribute(Qt::AA_EnableHighDpiScaling, true);
    // HighDPI scale factor policy must be set before QGuiApplication is created
    if (qgetenv("QT_SCALE_FACTOR_ROUNDING_POLICY").isEmpty())
        Application::setHighDpiScaleFactorRoundingPolicy(Qt::HighDpiScaleFactorRoundingPolicy::PassThrough);
#endif
```


This is Okular approach.
```
    /**
     * enable high dpi support
     */
    QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);

    /**
     * allow fractional scaling
     * we only activate this on Windows, it seems to creates problems on unices
     * (and there the fractional scaling with the QT_... env vars as set by KScreen works)
     * see bug 416078
     */
#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0) && defined(Q_OS_WIN)
    QGuiApplication::setHighDpiScaleFactorRoundingPolicy(Qt::HighDpiScaleFactorRoundingPolicy::PassThrough);
#endif
```

**Comment by pawelsalawa** [comment]
Interesting. Thanks for the explanation. I appreciate!
Also thank you for code snippets. I'll try it out.

**Comment by pawelsalawa** [comment]
@regs01 Did you mean earlier, that if you `set QT_SCALE_FACTOR_ROUNDING_POLICY=PassThrough` using current, unchanged binary of SQLiteStudio 3.3.0, then everything seems to be looking fine for you?

As far as I'm testing this under VirtualBox, where I have this 3400x2100 resolution and 175% scaling in Windows, the SQLiteStudio looks like this:
![image](https://user-images.githubusercontent.com/481846/108700077-5fa78600-7506-11eb-89f6-e5ccfec76ca0.png)
...and it doesn't matter if I set this environment variable (PassThrough). It always looks the same and is basically very much wrong (different label sizes, unreadable, small fonts).

I'd like to try out with your setup - what resolution and scaling do you have?

**Comment by regs01** [comment]
I'm having 4K 27" 150%. 

Checked. PassThrough properly scaling UI and most font, but some font indeed remain always unscaled. It's unnoticeable on 150%, but becomes noticable on higher scale. Likely something up to those properties of those controls.

Also notice panel headers are always huge, even at 96 DPI. Detach and close images seem to be oversized. Doesn't look to be DPI scaling related as well. Likely up to properties of controls.

I'm not into Qt, so I don't know if there are any scale related properties and any different font size units, scalable or not.

**Comment by pawelsalawa** [comment]
The scaling may have something to do with Fonts being configurable in the up. Perhaps I need to remove font customization and leave it up to Qt & the OS. Just like I did with colors customization (to address issues with Dark themes).

**Comment by regs01** [comment]
You can see UI scales properly. Most font also scales properly. But Configuration tree is scaled properly. DB tree won't scale and likely locked to some specific units, like pixels may be. As far as I remember GTK or Qt had unit types. And pixels would always remain exact, while points or em will scale. But not sure which one. Not in both those GTK or Qt. Just seen it somewhere few years ago.

175%
<img width="845" alt="SQLiteStudio_2021-02-22_14-22-23" src="https://user-images.githubusercontent.com/10468217/108702500-68a25280-751a-11eb-9682-237efdbe9c0c.png">

200%
<img width="965" alt="SQLiteStudio_2021-02-22_14-23-27" src="https://user-images.githubusercontent.com/10468217/108702512-6c35d980-751a-11eb-91dc-f97722be68db.png">

**Comment by pawelsalawa** [comment]
I've applied changes we discussed here. Still, there are numerous places, where fonts are too small if you pick 150% scale or bigger. This is a general problem in Qt (https://bugreports.qt.io/browse/QTBUG-89755 , https://bugreports.qt.io/browse/QTBUG-80228 and many, many more bugs reported for Qt). The situation will probably improve as new Qt versions are released and problems are addressed there.

I've compared these problems with qBittorrent and it's exactly the same situation there.

Anyway, with current improvements, the application looks better - as you recommended. Thank you for the contribution!
